MobileManager/Controllers/AppResourceController.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MobileManager.Controllers.Interfaces;
+using MobileManager.Http.Clients.Interfaces;
+using MobileManager.Logging.Logger;
+using MobileManager.Services.Interfaces;
+
+#pragma warning disable 1998
+
+namespace MobileManager.Controllers
+{
+    /// <inheritdoc cref="IAppResourceController" />
+    /// <summary>
+    /// App resources controller.
+    /// </summary>
+    [Route("api/v1/appResource")]
+    [EnableCors("AllowAllHeaders")]
+    public class AppResourceController : ControllerExtensions, IAppResourceController
+    {
+        private readonly IRestClient _restClient;
+        private readonly IManagerLogger _logger;
+        private readonly IAppResourceService _appResourceService;
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:MobileManager.Controllers.AppResourceController" /> class.
+        /// </summary>
+        /// <param name="restClient">Rest client.</param>
+        /// <param name="appResourceService">AppResourceService.</param>
+        /// <param name="logger">Logger.</param>
+        public AppResourceController(IRestClient restClient, IAppResourceService appResourceService, IManagerLogger logger) : base(logger)
+        {
+            _restClient = restClient;
+            _logger = logger;
+            _appResourceService = appResourceService;
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Gets the App resource info by hash identifier.
+        /// </summary>
+        /// <returns>The App resource by hash identifier.</returns>
+        /// <param name="id">Id hash.</param>
+        [HttpGet("{id}")]
+        public IActionResult GetById(string id)
+        {
+            LogRequestToDebug();
+            var appInfo = _appResourceService.FindAppResource(id);
+            
+            if (appInfo == null)
+            {
+                return NotFoundExtension("Application resource not found on storage.");
+            }
+
+            return JsonExtension(appInfo);
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Uploads from local storage APK/IPA source file for testing native applications
+        /// </summary>
+        /// <returns>Information about uploaded file</returns>
+        /// <param name="file">Source file.</param>
+        /// <response code="200">Application uploaded successfully</response>
+        /// <response code="400">Invalid upload request</response>
+        /// <response code="500">Internal failure</response>
+        [HttpPost("upload")]
+        public async Task<IActionResult> OnPostUploadAsync(IFormFile file)
+        {
+            LogRequestToDebug();
+            if (file == null || file.Length == 0)

```

Comment:
could you please you proper blocks with { }

Reply:
Will be fixed